src/Microsoft.OData.Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -611,7 +611,7 @@ private void MaterializeResolvedEntry(MaterializerEntry entry, bool includeLinks
                     }
 
                     // includeLinks is for Navigation property, so we should handle complex property when includeLinks equals false;
-                    if (!includeLinks && (prop.IsEntityCollection || prop.EntityCollectionItemType != null))
+                    if (!includeLinks && prop.EdmProperty is IEdmNavigationProperty)

```

Comment:
I do not understand this change..Can we have a navigation property if `includeLinks` is false?